plugins/aggregators/histogram/histogram.go

```diff
@@ -160,8 +183,18 @@ func (h *HistogramAggregator) Add(in telegraf.Metric) {
 // Push returns histogram values for metrics
 func (h *HistogramAggregator) Push(acc telegraf.Accumulator) {
 	metricsWithGroupedFields := []groupedByCountFields{}
+	now := time.Time{}
 
-	for _, aggregate := range h.cache {
+	if h.ExpirationInterval != 0 {
+		now = timeNow()
+	}
```

Comment:
I would just unconditionally do
```golang
now := timeNow()
```
here.

Reply:
But it affects performance a little bit. Why do we need to call `timeNow()` if we will not use it?

Reply:
Because it makes code simpler. There is always a trade-off between maintainability and performance. I don't know what a call to `time.Now()` costs, but given that this change saves use a few places we stray your new option to, I guess it's worth to take the "performance hit".

Reply:
Make sense, also I checked `time.Now()` performance and it's super fast